lib/UI.js

```diff
@@ -108,17 +109,36 @@ define(function (require, exports, module) {
             // If it is, the attached preview is hidden
             // and the detached preview is opened.
             if(Resizer.isVisible("#second-pane")) {
-                Resizer.hide("#second-pane");
-                $("#first-pane").addClass("expandEditor");
-                IframeBrowser.detachPreview();
+                _detachPreview();
             }
             else {
-                Resizer.show("#second-pane");
-                $("#first-pane").removeClass("expandEditor");
+                _attachPreview();
             }
         });
     }
 
+    /**
+      * Used to hide second pane, spawn detached preview, and attach beforeunload listener
+      */
+    function _detachPreview() {
+        Resizer.hide("#second-pane");
+        $("#first-pane").addClass("expandEditor");
+        $('#liveDevButton').removeClass("liveDevButton");
+        $('#liveDevButton').addClass("liveDevButtonAlt");
+        IframeBrowser.detachPreview();
```

Comment:
Let's make `IframeBrowser.detachPreview()` return `detachedWindow` so you can rewrite this code like so:

```js
var detachedWindow = IframeBrowser.detachPreview();
detachedPreview.addEventListener("beforeunload", _attachPreview, false);
```

Also note that I've just used passed the `_attachPreview` function directly vs. wrapping in a new function.